pkg/metrics/metrics.go

```diff
@@ -179,10 +180,25 @@ type SendMetricRequest struct {
 	InstallID string `json:"installId"`
 }
 
-// WriteMetric sends information about application usage. Noop if metrics
-// are opted out.
+// WriteMetric calls WriteMetricSync in a go routine.
+// It returns a closure to be run after program logic which will block until a response is returned
+// or provided context is canceled. If no provided deadline in context, defaults to 2 seconds.
+func (c *client) WriteMetric(ctx context.Context, name string, count int64) func() {
+	cancel := func() {}
```

Comment:
What is this for?

Reply:
I only make a new context if there isn't an existing timeout.

In the case I make a new context I need to call `cancel()` once the context is no longer needed. I decided to have a noop in the case I didn't make a new context, though I could also check for nil if that feels more idomatic

Reply:
I went ahead and merged in your suggestion that just always has a deadline of 5 seconds, though the one scenario where I could see that not being desired is if you were debugging metrics and wanted to set a very long timeout so you can include breakpoints.